library/src/main/java/com/datatorrent/lib/io/fs/FilterStreamCodec.java

```diff
@@ -108,4 +115,71 @@ public void reclaimFilterStreamContext(FilterStreamContext<CipherOutputStream> f
 
     }
   }
+
+  public static class SnappyFilterStream extends FilterOutputStream
+  {
+    /**
+     * Creates an output stream filter built on top of the specified
+     * underlying output stream.
+     *
+     * @param out the underlying output stream to be assigned to
+     *            the field <tt>this.out</tt> for later use, or
+     *            <code>null</code> if this instance is to be
+     *            created without an underlying stream.
+     */
+    public SnappyFilterStream(CompressionOutputStream out)
+    {
+      super(out);
+    }
+
+    public void finish() throws IOException
+    {
+      ((CompressionOutputStream)out).finish();
+    }
+  }
+
+  public static class SnappyFilterStreamContext extends FilterStreamContext.BaseFilterStreamContext<SnappyFilterStream>
+  {
+    public SnappyFilterStreamContext(OutputStream outputStream) throws IOException
+    {
+      SnappyCodec codec = new SnappyCodec();
```

Comment:
This check actually already happens inside of SnappyCodec. I add it inside the tests so that the tests don't run and fail but do display the appropriate warning